### ISSUE TITLE
macOS ARM64 support (Apple Silicon)

### ISSUE THREAD

**Issue by Tarik02** [OpenSAGE/OpenSAGE#598]

**Comment by Tarik02** [comment]
This requires migration to .NET 6.0

**Comment by Tarik02** [comment]
Could successfully build arm64 Launcher on arm64 machine using .NET SDK 6.0.100-preview.7 by applying this patch:
```
diff --git a/src/Directory.Build.props b/src/Directory.Build.props
index 65bf98d4..ef85b6be 100644
--- a/src/Directory.Build.props
+++ b/src/Directory.Build.props
@@ -5,4 +5,8 @@
     <ImGuiVersion>5.72</ImGuiVersion>
     <SharpAudioVersion>1.0.7-beta</SharpAudioVersion>
   </PropertyGroup>
+
+  <PropertyGroup>
+    <ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles>
+  </PropertyGroup>
 </Project>
\ No newline at end of file
diff --git a/src/OpenSage.Launcher/OpenSage.Launcher.csproj b/src/OpenSage.Launcher/OpenSage.Launcher.csproj
index ad88adc4..7f58be26 100644
--- a/src/OpenSage.Launcher/OpenSage.Launcher.csproj
+++ b/src/OpenSage.Launcher/OpenSage.Launcher.csproj
@@ -2,8 +2,8 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net5.0</TargetFramework>
-    <RuntimeIdentifiers>win-x64;osx-x64;ubuntu-x64</RuntimeIdentifiers>
+    <TargetFramework>net6.0</TargetFramework>
+    <RuntimeIdentifiers>win-x64;osx-x64;osx-arm64;ubuntu-x64</RuntimeIdentifiers>
     <ApplicationIcon>Resources\AppIcon.ico</ApplicationIcon>
     <ApplicationManifest>Resources\app.manifest</ApplicationManifest>
   </PropertyGroup>
```

```
$ file build/artifacts/osx-arm64/OpenSage.Launcher
build/artifacts/osx-arm64/OpenSage.Launcher: Mach-O 64-bit executable arm64
```

**Comment by Tarik02** [comment]
Seems like it uses x86_64 version glsllangValidator using Rosetta 2. Would be cool to find arm64 version of that binary.

**Comment by feliwir** [comment]
Could you submit the patch as PR?

**Comment by Tarik02** [comment]
> Could you submit the patch as PR?

Even if I will do it, it is still not ready to be merged to master since it is blocked by release of .NET Core 6. But I will definitely do it.

**Comment by Tarik02** [comment]
Could not successfully run it since Veldrid initialization fails: https://github.com/mellinoe/veldrid/issues/402

**Comment by Tarik02** [comment]
`PixelFormat.D24_UNorm_S8_UInt` is not supported on M1 GPU. Game crashes with:
```
validateWithDevice:2556: failed assertion `Render Pipeline Descriptor Validation
depthAttachmentPixelFormat is not a valid MTLPixelFormat.
depthAttachmentPixelFormat MTLPixelFormatInvalid is not depth renderable.
stencilAttachmentPixelFormat is not a valid MTLPixelFormat.
stencilAttachmentPixelFormat MTLPixelFormatInvalid is not stencil renderable.
'
```

This can be checked by reading `MTLDevice.isDepth24Stencil8PixelFormatSupported`.

**Comment by Tarik02** [comment]
After commenting out something in rendering code (replaced depth-stencil buffer format to another one) I could get this running:
<img width="1086" alt="Знімок екрана 2021-08-17 о 15 16 20" src="https://user-images.githubusercontent.com/12175048/129724039-e77877de-647e-4ec5-bb16-4421971f1043.png">

I even got audio working! Sadly, the game view is empty. This seems to be some Veldrid bug around Metal bindings (other renderers do not work: OpenGL on macOS is outdated, MoltenVK seems not to work well (some bug around memory allocation)).

**Comment by feliwir** [comment]
@Tarik02 I think Veldrid should probably have a way to check for supported pixelformats. We're usually not having checks like these directly in OpenSAGE (we won't make any MTL calls ourselves)

**Comment by Tarik02** [comment]
@feliwir yes, of course. As I see, Veldrid has `GraphicsDevice.GetPixelFormatSupport` function.

I think we should consider possibility to work without `D24_UNorm_S8_UInt` support because is often missing on mobile GPUs (including Apple M1 chip).

**Comment by feliwir** [comment]
Alright! Regarding the pink screen: I think it's because our RenderTarget could not be initialized, which should render in there. Probably also due to the missing format support: https://github.com/OpenSAGE/OpenSAGE/blob/f0fd3ee295328b7060e7452f9ef24a6076bca214/src/OpenSage.Game/Graphics/Rendering/RenderPipeline.cs#L23

**Comment by Tarik02** [comment]
@feliwir I don't think that this is problem with OpenSAGE at all. I got this problem with Veldrid.NeoDemo project and there seems to be no problems with formats (tracking at https://github.com/mellinoe/veldrid/issues/402).

**Comment by Tarik02** [comment]
Also, there's an interesting thing to notice: OpenSAGE has working UI while pink game view, while that example project has just pink screen. ImGUI is rendered via Metal, am I right?

**Comment by feliwir** [comment]
Yes, it is when you have the metal renderer selected. We don't switch backends within OpenSAGE

**Comment by Tarik02** [comment]
Pink screen seems to be caused by this bug: https://github.com/mellinoe/veldrid/issues/289
